File: tests/test-clients/java-okhttp-jackson/src/test/java/test_client/CheckClientTest.java
package test_client;

import org.junit.jupiter.api.Test;
import test_client.clients.check.CheckClient;
import test_client.errors.ClientException;

import static org.junit.jupiter.api.Assertions.*;

import static test_client.Constants.*;

public class CheckClientTest {
	private final CheckClient client = new CheckClient(BASE_URL);

	@Test
	public void checkEmpty_doesntThrowException() {
		assertDoesNotThrow(client::checkEmpty);
	}

	@Test
	public void checkEmptyResponse_doesntThrowException() {
		assertDoesNotThrow(() -> client.checkEmptyResponse(MESSAGE));
	}

	@Test
	public void checkForbidden_doesntThrowException() {
		assertThrows(ClientException.class, client::checkForbidden);
	}
}

File: test-clients/kotlin-micronaut-jackson/src/test/kotlin/test_client/CheckClientTest.kt
package test_client

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import test_client.clients.check.CheckClient
import test_client.errors.*
import test_client.errors.models.*
import test_client.models.*

class CheckClientTest {
    private val baseUrl = "http://localhost:8081"
    private val client = CheckClient(baseUrl)

    private val bodyMessage = Message(123, "the string")

    @Test
    fun checkEmpty_doesntThrowException() {
        assertDoesNotThrow { client.checkEmpty() }
    }

    @Test
    fun checkEmptyResponse_doesntThrowException() {
        assertDoesNotThrow { client.checkEmptyResponse(bodyMessage) }
    }

    @Test
    fun checkForbidden_throwException() {
        val exception = assertThrows<ClientException> { client.checkForbidden() }
        assertTrue(exception.cause is ForbiddenException)
    }

    @Test
    fun checkConflict_checkExceptionBody() {
        val exception = assertThrows<ClientException> { client.checkConflict() }.cause as ConflictException
        val expectedBody = ConflictMessage("Conflict with the current state of the target resource")
        assertEquals(exception.body, expectedBody)
    }

    @Test
    fun checkConflict_throwException() {
        val exception = assertThrows<ClientException> { client.checkConflict() }
        assertTrue(exception.cause is ConflictException)
    }

    @Test
    fun checkBadRequest_checkExceptionBody() {
        val exception = assertThrows<ClientException> { client.checkBadRequest() }.cause as BadRequestErrorException
        val expectedBody = BadRequestError("Failed to execute request", ErrorLocation.UNKNOWN, null)
        assertEquals(exception.body, expectedBody)
    }

    @Test
    fun checkBadRequest_throwException() {
        val exception = assertThrows<ClientException> { client.checkBadRequest() }
        assertTrue(exception.cause is BadRequestErrorException)
    }
}